frontend/src/views/Game.tsx

```diff
@@ -20,57 +20,80 @@ type LocationState = {
 function GamePage() {
   const location = useLocation<LocationState>();
   const [room, setRoom] = useState<Room | null>(null);
-  const [problems, setProblems] = useState<Problem[]>([]);
+  const [problem, setProblem] = useState<Problem | null>(null);
+  const [submission, setSubmission] = useState<SubmissionResult | null>(null);
   const [error, setError] = useState<string>('');
 
   // Called every time location changes
   useEffect(() => {
     if (checkLocationState(location, 'room')) {
       setRoom(location.state.room);
     }
-    getProblems().then((res) => {
-      if (!(res as Problem[]).length) {
-        setError('Problem cannot be found');
-      } else {
-        setProblems(res as Problem[]);
-        setError('');
-      }
-    }).catch((err) => {
-      setError((err as ErrorResponse).message);
-      setProblems([]);
-    });
+    getProblems()
+      .then((res) => {
+        if (!res.length) {
+          setError('Problem cannot be found');
+        } else {
+          setProblem(res[0]);
+          setError('');
+        }
+      }).catch((err) => setError(err.message));
   }, [location]);
 
-  const firstProblem = problems?.[0];
-
   // Creates Event when splitter bar is dragged
   const onSecondaryPanelSizeChange = () => {
     const event = new Event('secondaryPanelSizeChange');
     window.dispatchEvent(event);
   };
 
+  // Callback when user runs code against custom test case
+  const runSolution = (input: string) => {
+    // Needs to be implemented
```

Comment:
nit: Maybe add "TODO" if you're going to keep this in? Or just remove the comment?